Python/pylifecycle.c

```diff
@@ -350,24 +350,25 @@ _coerce_default_locale_settings(int warn, const _LocaleCoercionTarget *target)
                 "Error setting LC_CTYPE, skipping C locale coercion\n");
         return;
     }
-    if (warn) {
-        fprintf(stderr, C_LOCALE_COERCION_WARNING, newloc);
-    }
 
     /* Reconfigure with the overridden environment variables */
     _Py_SetLocaleFromEnv(LC_ALL);
 }
 #endif
 
-void
-_Py_CoerceLegacyLocale(int warn)
+int
+_Py_CoerceLegacyLocale(const char **coercion_target,
+                       const char **coercion_warning)
 {
+    int locale_was_coerced = 0;
+    *coercion_target = NULL;
```

Comment:
This is missing checks for `coercion_target` and `coercion_warning` being NULL, and skipping the assignments in that case.

Reply:
I added these missing checks in https://github.com/python/cpython/pull/9257/commits/f4c686babca9870fc832797f2b960a6420c14f38